app/code/core/Mage/Shipping/Model/Config.php

```diff
@@ -102,6 +102,11 @@ protected function _getCarrier($code, $config, $store = null)
         }
         $modelName = $config['model'];
 
+        $className = Mage::getConfig()->getModelClassName($modelName);
+        if (!mageFindClassFile($className)) {
+            return false;
+        }
+
         /**
          * Added protection from not existing models usage.
          * Related with module uninstall process
```

Comment:
**Discussion**
If the model exists, it will call `getModelClassName` twice. `getModel()` check for `class_exists`:
https://github.com/OpenMage/magento-lts/blob/59629098f42ce7d4e2052c989f3fbd5523fb2ace/app/code/core/Mage/Core/Model/Config.php#L1382-L1385
 `mageFindClassFile` checks for `file_exists`:
https://github.com/OpenMage/magento-lts/blob/59629098f42ce7d4e2052c989f3fbd5523fb2ace/app/code/core/Mage/Core/functions.php#L90-L96. 

But I think in this case, lines 105 to 120 (including `try catch` block) can be simplified to

```suggestion
            /** @var Mage_Shipping_Model_Carrier_Abstract $carrier */
            $carrier = Mage::getModel($modelName);
            if (!$carrier) {
                return false;
            }
```

The same can be applied here too:
https://github.com/OpenMage/magento-lts/blob/59629098f42ce7d4e2052c989f3fbd5523fb2ace/app/code/core/Mage/Payment/Model/Config.php#L92-L97

What do you think?

Reply:
I made the proposed changes :)